include/deal.II/particles/particle_handler.h

```diff
@@ -86,7 +86,10 @@ namespace Particles
     /**
      * Destructor.
      */
-    virtual ~ParticleHandler() override = default;
+    virtual ~ParticleHandler() override
+    {
+      connection.disconnect();
```

Comment:
Could you use a `boost::signals2::scoped_connection` instead of a `boost::signals2::connection` and avoid creating this destructor? I have not used a scoped connection before, but it looks like it is made for cases like this.